vm/x86/x86defs/src/vmx.rs

```diff
@@ -290,6 +292,120 @@ pub struct InterruptionInformation {
     pub valid: bool,
 }
 
+#[bitfield(u32)]
+pub struct GdtrOrIdtrInstructionInfo {
+    #[bits(2)]
+    pub scaling: u8,
+    #[bits(5)]
+    _reserved1: u8,
+    #[bits(3)]
+    pub address_size: u8,
+    _reserved2: bool,
+    pub operand_size: bool,
+    #[bits(3)]
+    _reserved3: u8,
+    #[bits(3)]
+    pub segment_register: u8,
+    #[bits(4)]
+    pub index_register: u8,
+    pub index_register_invalid: bool,
+    #[bits(4)]
+    pub base_register: u8,
+    pub base_register_invalid: bool,
+    #[bits(2)]
+    pub instruction: GdtrOrIdtrInstruction,
+    #[bits(2)]
+    _reserved4: u8,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum GdtrOrIdtrInstruction {
+    Sgdt = 0,
+    Sidt = 1,
+    Lgdt = 2,
+    Lidt = 3,
+}
+
+impl GdtrOrIdtrInstruction {
+    pub const fn from_bits(value: u8) -> Self {
```

Comment:
I don't like making `from_bits` `pub` when the mapping from the input type is partial like this.